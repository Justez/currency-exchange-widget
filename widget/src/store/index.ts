import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import createRootReducer from './root-reducer';
import sagas from './root-saga';

const configureStore = () => {
  const sagaMiddleware = createSagaMiddleware();

  const middlewares = [
    sagaMiddleware,
  ];

  const instance = createStore(
    createRootReducer(),
    composeWithDevTools(applyMiddleware(...middlewares)),
  );

  sagaMiddleware.run(sagas);

  return instance;
};

export type State = ReturnType<typeof store.getState>;

const store = configureStore();

export default store;
