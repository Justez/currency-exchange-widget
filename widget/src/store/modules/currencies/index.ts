//@ts-ignore
import { createActions } from 'redux-actions';

export const NAMESPACE = 'currencies';

export const actions = createActions(
  'SET_SELECTED_CURRENCY',
  'FLIP_SELECTED_CURRENCIES',
  { prefix: NAMESPACE },
);
