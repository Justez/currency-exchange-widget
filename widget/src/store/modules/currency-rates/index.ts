import { createActions } from 'redux-actions';

export const NAMESPACE = 'currencyRates';

export const actions = createActions(
  'FLIP_RATES',
  'GET_CURRENCY_RATES',
  'GET_CURRENCY_RATES_REQUEST',
  'GET_CURRENCY_RATES_SUCCESS',
  'GET_CURRENCY_RATES_FAILURE',
  'SET_CURRENCY_RATES',
  'RESET_RATES',
  { prefix: NAMESPACE },
);
