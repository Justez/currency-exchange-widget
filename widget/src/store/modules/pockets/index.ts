import { createActions } from 'redux-actions';

export const NAMESPACE = 'pockets';

export const actions = createActions(
  'PLACE_SUM',
  'PLACE_ALL',
  'PLACE_EXCHANGE',
  'PLACE_EXCHANGE_REQUEST',
  'PLACE_EXCHANGE_SUCCESS',
  'PLACE_EXCHANGE_FAILURE',
  'SET_PLACED_AMOUNT',
  { prefix: NAMESPACE },
);
