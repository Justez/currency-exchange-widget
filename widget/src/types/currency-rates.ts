import { CurrencyMap } from 'types';

export type CurrencyRateResponse = {
  data: {
    base: string;
    rates: {
      [key in CurrencyMap]: number;
    };
  }
}

export type CurrencyRateError = {
    error: boolean;
    status: number;
    message: string;
    description: string;
}