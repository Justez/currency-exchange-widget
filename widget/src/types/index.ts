export * from './rates'
export * from './currencies'
export * from './pockets'