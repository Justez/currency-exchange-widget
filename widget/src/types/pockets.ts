import { CurrencyMap } from './'

export type Pocket = {
    currency: CurrencyMap,
    sum: string;
    placedSum: string;
}

export type Pockets = Array<Pocket>