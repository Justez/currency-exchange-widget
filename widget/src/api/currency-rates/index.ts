import axios from 'axios'

import { Currencies,CurrencyRateResponse, CurrencyRateError } from 'types';

const URL = 'https://openexchangerates.org/api/latest.json';

export const getCurrencyRates = (obj: Currencies) => axios.get(URL, {
    params: {
        app_id: '2b3e9170d55c46fead589f4fdeac1fb8',
        // base: obj.out.toUpperCase(),
    }
})
    .then((res: CurrencyRateResponse) => res.data)
    .catch((err: CurrencyRateError) => err);