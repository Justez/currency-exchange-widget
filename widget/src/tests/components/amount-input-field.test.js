import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

import store from 'store';
import AmountInputField from 'components/amount-input-field';
import { actions } from 'store/modules/currency-rates';
import { actions as pocketActions } from 'store/modules/pockets';

import { CurrencyExchangeTypes } from 'types';

const component = (
    <Provider store={store}>
        <AmountInputField pocketDirection={CurrencyExchangeTypes.out} />
        <AmountInputField pocketDirection={CurrencyExchangeTypes.in} />
    </Provider>
)

let wrapper;
let firstInput, secondInput;

describe('AmountInputField input integration', () => {
    beforeAll(() => {
        store.dispatch(actions.setCurrencyRates({ rate: 1.22, reverse: 0.88 }));
        wrapper = mount(component);
        firstInput = wrapper.find('input').first();
        secondInput = wrapper.find('input').at(1);
    });

    it('input is calculated properly', () => {
        expect(firstInput.props().value).toBe(4);
    });

    it('withdraw sum is calculated properly', () => {
        expect(secondInput.props().value).toBe(4.88);
    });

    it('withdraw sum changed after input is editted', () => {
        store.dispatch(pocketActions.placeSum({ placedSum: 6, pocketDirection: CurrencyExchangeTypes.out }));
        store.dispatch(pocketActions.placeSum({ placedSum: 6, pocketDirection: CurrencyExchangeTypes.out }));
        wrapper = mount(component);
        secondInput = wrapper.find('input').at(1)
        expect(secondInput.props().value).toBe(7.32);
    });

    it('input sum changed after input is editted', () => {
        store.dispatch(pocketActions.placeSum({ placedSum: 6, pocketDirection: CurrencyExchangeTypes.in }));
        wrapper = mount(component);
        firstInput = wrapper.find('input').first()
        expect(firstInput.props().value).toBe(5.28);
    });
});

describe('AmountInputField snapshots', () => {
    const render = () => renderer.create(component).toJSON();

    describe('initial state', () => {
        it('renders out/in inputs', () => {
            expect(render()).toMatchSnapshot();
        });
    });

    describe('rates are present', () => {
        beforeAll(() => {
            store.dispatch(actions.setCurrencyRates({ rate: 1.22, reverse: 0.88 }))
        });

        it('renders out/in inputs with updated store', () => {
            act(() => {
                store.dispatch(pocketActions.placeSum({ placedSum: 6, pocketDirection: CurrencyExchangeTypes.out }));
            });
            expect(render()).toMatchSnapshot();
        });

        it('renders out/in inputs with updated withdraw sum', () => {
            act(() => {
                store.dispatch(pocketActions.placeSum({ placedSum: 6, pocketDirection: CurrencyExchangeTypes.in }));
            });
            expect(render()).toMatchSnapshot();
        });

        it('renders out/in inputs with updated withdraw sum', () => {
            act(() => {
                store.dispatch(pocketActions.placeSum({ placedSum: 100, pocketDirection: CurrencyExchangeTypes.in }));
            });
            expect(render()).toMatchSnapshot();
        });

        it('renders correct out/in inputs if exceeding sum was placed', () => {
            act(() => {
                store.dispatch(pocketActions.placeSum({ placedSum: 150, pocketDirection: CurrencyExchangeTypes.in }));
            });
            expect(render()).toMatchSnapshot();
        });
    });
});

export default component;