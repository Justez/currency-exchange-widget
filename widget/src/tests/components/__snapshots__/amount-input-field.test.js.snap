// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountInputField snapshots initial state renders out/in inputs 1`] = `
Array [
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13H5v-2h14v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-input"
      max={868.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={5.28}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "15px",
            "marginBottom": "-2px",
            "width": "15px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5h5.14c.38 0 .73-.21.89-.55.33-.66-.15-1.45-.89-1.45h-5.8c-.05-.33-.08-.66-.08-1s.03-.67.08-1h5.8c.38 0 .73-.21.89-.55.34-.67-.14-1.45-.89-1.45H9.24C10.32 6.92 12.5 5.5 15 5.5c1.25 0 2.42.36 3.42.97.5.31 1.15.26 1.57-.16.58-.58.45-1.53-.25-1.96C18.36 3.5 16.73 3 15 3c-3.92 0-7.24 2.51-8.48 6h-2.9c-.38 0-.73.21-.9.55-.33.67.15 1.45.9 1.45h2.44c-.04.33-.06.66-.06 1s.02.67.06 1H3.62c-.38 0-.73.21-.89.55-.34.67.14 1.45.89 1.45h2.9c1.24 3.49 4.56 6 8.48 6 1.74 0 3.36-.49 4.74-1.35.69-.43.82-1.39.24-1.97-.42-.42-1.07-.47-1.57-.15-.99.62-2.15.97-3.41.97z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-received"
      max={102.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={6}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "11",
            "marginLeft": "4px",
            "width": "11",
          }
        }
        viewBox="0 0 8 8"
      >
        <path
          d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`AmountInputField snapshots rates are present renders correct out/in inputs if exceeding sum was placed 1`] = `
Array [
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13H5v-2h14v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-input"
      max={868.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={89.95}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "15px",
            "marginBottom": "-2px",
            "width": "15px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5h5.14c.38 0 .73-.21.89-.55.33-.66-.15-1.45-.89-1.45h-5.8c-.05-.33-.08-.66-.08-1s.03-.67.08-1h5.8c.38 0 .73-.21.89-.55.34-.67-.14-1.45-.89-1.45H9.24C10.32 6.92 12.5 5.5 15 5.5c1.25 0 2.42.36 3.42.97.5.31 1.15.26 1.57-.16.58-.58.45-1.53-.25-1.96C18.36 3.5 16.73 3 15 3c-3.92 0-7.24 2.51-8.48 6h-2.9c-.38 0-.73.21-.9.55-.33.67.15 1.45.9 1.45h2.44c-.04.33-.06.66-.06 1s.02.67.06 1H3.62c-.38 0-.73.21-.89.55-.34.67.14 1.45.89 1.45h2.9c1.24 3.49 4.56 6 8.48 6 1.74 0 3.36-.49 4.74-1.35.69-.43.82-1.39.24-1.97-.42-.42-1.07-.47-1.57-.15-.99.62-2.15.97-3.41.97z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-received"
      max={102.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={102.22}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "11",
            "marginLeft": "4px",
            "width": "11",
          }
        }
        viewBox="0 0 8 8"
      >
        <path
          d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`AmountInputField snapshots rates are present renders out/in inputs with updated store 1`] = `
Array [
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13H5v-2h14v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-input"
      max={868.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={6}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "15px",
            "marginBottom": "-2px",
            "width": "15px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5h5.14c.38 0 .73-.21.89-.55.33-.66-.15-1.45-.89-1.45h-5.8c-.05-.33-.08-.66-.08-1s.03-.67.08-1h5.8c.38 0 .73-.21.89-.55.34-.67-.14-1.45-.89-1.45H9.24C10.32 6.92 12.5 5.5 15 5.5c1.25 0 2.42.36 3.42.97.5.31 1.15.26 1.57-.16.58-.58.45-1.53-.25-1.96C18.36 3.5 16.73 3 15 3c-3.92 0-7.24 2.51-8.48 6h-2.9c-.38 0-.73.21-.9.55-.33.67.15 1.45.9 1.45h2.44c-.04.33-.06.66-.06 1s.02.67.06 1H3.62c-.38 0-.73.21-.89.55-.34.67.14 1.45.89 1.45h2.9c1.24 3.49 4.56 6 8.48 6 1.74 0 3.36-.49 4.74-1.35.69-.43.82-1.39.24-1.97-.42-.42-1.07-.47-1.57-.15-.99.62-2.15.97-3.41.97z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-received"
      max={102.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={7.32}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "11",
            "marginLeft": "4px",
            "width": "11",
          }
        }
        viewBox="0 0 8 8"
      >
        <path
          d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`AmountInputField snapshots rates are present renders out/in inputs with updated withdraw sum 1`] = `
Array [
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13H5v-2h14v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-input"
      max={868.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={5.28}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "15px",
            "marginBottom": "-2px",
            "width": "15px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5h5.14c.38 0 .73-.21.89-.55.33-.66-.15-1.45-.89-1.45h-5.8c-.05-.33-.08-.66-.08-1s.03-.67.08-1h5.8c.38 0 .73-.21.89-.55.34-.67-.14-1.45-.89-1.45H9.24C10.32 6.92 12.5 5.5 15 5.5c1.25 0 2.42.36 3.42.97.5.31 1.15.26 1.57-.16.58-.58.45-1.53-.25-1.96C18.36 3.5 16.73 3 15 3c-3.92 0-7.24 2.51-8.48 6h-2.9c-.38 0-.73.21-.9.55-.33.67.15 1.45.9 1.45h2.44c-.04.33-.06.66-.06 1s.02.67.06 1H3.62c-.38 0-.73.21-.89.55-.34.67.14 1.45.89 1.45h2.9c1.24 3.49 4.56 6 8.48 6 1.74 0 3.36-.49 4.74-1.35.69-.43.82-1.39.24-1.97-.42-.42-1.07-.47-1.57-.15-.99.62-2.15.97-3.41.97z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-received"
      max={102.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={6}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "11",
            "marginLeft": "4px",
            "width": "11",
          }
        }
        viewBox="0 0 8 8"
      >
        <path
          d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
        />
      </svg>
    </div>
  </div>,
]
`;

exports[`AmountInputField snapshots rates are present renders out/in inputs with updated withdraw sum 2`] = `
Array [
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13H5v-2h14v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-input"
      max={868.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={88}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "15px",
            "marginBottom": "-2px",
            "width": "15px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5h5.14c.38 0 .73-.21.89-.55.33-.66-.15-1.45-.89-1.45h-5.8c-.05-.33-.08-.66-.08-1s.03-.67.08-1h5.8c.38 0 .73-.21.89-.55.34-.67-.14-1.45-.89-1.45H9.24C10.32 6.92 12.5 5.5 15 5.5c1.25 0 2.42.36 3.42.97.5.31 1.15.26 1.57-.16.58-.58.45-1.53-.25-1.96C18.36 3.5 16.73 3 15 3c-3.92 0-7.24 2.51-8.48 6h-2.9c-.38 0-.73.21-.9.55-.33.67.15 1.45.9 1.45h2.44c-.04.33-.06.66-.06 1s.02.67.06 1H3.62c-.38 0-.73.21-.89.55-.34.67.14 1.45.89 1.45h2.9c1.24 3.49 4.56 6 8.48 6 1.74 0 3.36-.49 4.74-1.35.69-.43.82-1.39.24-1.97-.42-.42-1.07-.47-1.57-.15-.99.62-2.15.97-3.41.97z"
        />
      </svg>
    </div>
  </div>,
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-root-1 MuiInputBase-adornedStart MuiInputBase-adornedEnd"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      style={
        Object {
          "height": "18px",
        }
      }
      viewBox="0 0 24 24"
    >
      <path
        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
      />
    </svg>
    <input
      className="MuiInputBase-input MuiInput-input makeStyles-input-2 MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd"
      disabled={false}
      id="amount-received"
      max={102.22}
      maxLength={7}
      min={0}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="number"
      value={100}
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        style={
          Object {
            "height": "11",
            "marginLeft": "4px",
            "width": "11",
          }
        }
        viewBox="0 0 8 8"
      >
        <path
          d="M3 0c-.62 0-1.16.26-1.5.69-.34.43-.5.99-.5 1.56 0 .69.16 1.25.25 1.75h-1.25v1h1.22c-.11.45-.37.96-1.06 1.66l-.16.13v1.22h6v-1h-4.91c.64-.73.98-1.4 1.13-2h1.78v-1h-1.72c-.08-.68-.28-1.24-.28-1.75 0-.39.11-.73.28-.94.17-.21.37-.31.72-.31.39 0 .61.11.75.25s.25.36.25.75h1c0-.58-.17-1.1-.53-1.47-.37-.37-.89-.53-1.47-.53z"
        />
      </svg>
    </div>
  </div>,
]
`;
